pkg/cluster/sync.go

```diff
@@ -118,6 +118,10 @@ func (c *Cluster) Sync(newSpec *acidv1.Postgresql) error {
 		return fmt.Errorf("could not sync connection pooler: %v", err)
 	}
 
+	if err := c.majorVersionUpgrade(); err != nil {
```

Comment:
Is there an implicit requirement that `c.majorVersionUpgrade()` must be called after all other `Sync` actions have completed successfully ?